packages/mobile/src/components/Chat/ChatDetail.tsx

```diff
@@ -888,6 +902,14 @@ const RenderMessageText = memo((props: MessageTextProps<IMessage>) => {
     ];
   }, []);
 
+  const onExpand = useCallback(() => {
+    if (!hasMoreTextContent || !completeMessage) return;
+    const message = props.currentMessage as ChatMessageIMessage;
+    message.text = completeMessage.message;
+    props.currentMessage = message;
```

Comment:
Editing a property from within the component is a big no-no I'm afraid. Why don't you pass the `currentMessage` prop into a state variable `message` so you can do a nice `setMessage` when you've fetched the extra payload